webmessaging/multi-globals/messageport-current.html

```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Making the current page become non-active must prevent message transmission</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<!-- This is the entry global -->
+
+<iframe src="support/incumbent.html" id="incumbent"></iframe>
+<iframe src="support/current.html" id="current"></iframe>
+
+<script>
+"use strict";
+setup({ explicit_done: true });
+
+const incumbentIframe = document.querySelector("#incumbent");
+const currentIframe = document.querySelector("#current");
+
+window.onload = () => {
+  promise_test(async t => {
+    const messageChannel = frames[0].createMessageChannel();
+
+    await new Promise((resolve, reject) => {
+      currentIframe.onload = () => resolve();
+      currentIframe.onerror = () => reject(new Error("Could not navigate the iframe"));
+      currentIframe.src = "/common/blank.html";
+    });
+
+    messageChannel.port1.onmessage = t.unreached_func("message event recieved");
+    messageChannel.port1.onmessageerror = t.unreached_func("messageerror event recieved");
+    messageChannel.port2.postMessage("boo");
+
+    await new Promise(resolve => t.step_timeout(resolve, 3000));
```

Comment:
If you create another messageChannel in the entry global and use that, can we expect anything about its speed relative to the other one? That might allow using less resources on CI.

Reply:
I don't believe so, because each MessagePort has its own independent message queue, with no ordering guarantees between them. (Unlike BroadcastChannel.)

Reply:
I think it would be good to have a comment here explaining the timeout so those looking into removing timeouts know roughly what kind of API would have to exist.